design/one-pager-template-stacks.md

```diff
@@ -0,0 +1,399 @@
+* [No Custom Controller Needed](#No-Custom-Controller-Needed)
+  
+  There is a single all-purpose controller that is designed to managed Template Stacks. The existing Stack Manager will wire the Template Stack Manager for managing the resources defined in a Template Stack.
+* [Composition](#Composition)
+
+  Stacks can be embeded in different Cluster environments, including other Stacks.  Charts and custom YAML files may reside in the cluster in the form of a Template Stack.
+
+  Template Stacks introduce template variables that are take advantage of the outputs of templated resources.  All through standard Kubernetes reconciliation.
+
+### Simple Variable Substitution
+
+A Crossplane Stack CR currently accepts a `spec.controller.deployment` parameter, which specifies a single Kubernetes `Deployment` resource as its value.
+
+```yaml
+kind: Stack
+metadata:
+  name: RedisStack
+spec:
+  owns:
+  - kind: Redis
+    apiVersion: redis.example.org/v1alpha1
+  controller:
+    deployment:
+      <typical Kubernetes Deployment resource>
+…
+```
+
+*The `deployment` value is populated from an `install.yaml` file contained within the Stack filesystem which the Stack Manager discovers.*
+
+If this Stack wanted to parameterize a piece of this `Deployment`, say the image version for a Redis container image, we could imagine placing a template variable in the `spec.controller.deployment` field.
+
+Supposing we used an inline handlebar template syntax, the `deployment` field could contain:
+
+```yaml
+containers:
+- name: redis-controller
+  image: redis/redis-crossplane-controller:{{.spec.version}}
+  imagePullPolicy: Always
+```
+
+_In practice, `version` would most likely be static, pinned to a specific version of a Stack_.
+
+In this example, the `.spec.version` value would be taken from the body of each CR this Stack manages. That value would be applied to the template and rendered as a new or updated `Deployment` resource.
+
+Should the `controller.deployment` value be treated as a template? This could have unintended consequences for existing Stack deployments, especially if their definitions includes pieces that match the template format.
+
+To avoid disturbing existing `controller.deployment` values, let's use a new template aware `templates` spec field. We'll take that approach in the following sections.
+
+### No Custom Controller Needed
+
+Existing Stacks must specify a `Deployment`. These are typically written in Go and generally require understanding some combination of client-go, controller-runtime, crossplane-runtime, and Kubebuilder.
+
+For some applications, the intended work of the controller could be simple:
+
+* For every new resource, create a set of Kubernetes resources
+* Use the spec fields of that resource to affect the Kubernetes resources
+* Upon changes to these spec fields, update the Kubernetes resources
+
+This is where the value of Template Stacks starts to materialize.
```

Comment:
I am thinking it would probably make sense to repeat the benefit at this point. Maybe something like:

```
This is where the value of Template Stacks starts to materialize. A stack author doesn't need to write their own controller, because a common one is provided by the Template Stacks Manager.
```

It might make sense to put the restatement of the benefit at the end of the first paragraph though.

I am thinking it makes sense to restate it so that the section is a bit more self-contained. It also helps me as a reader understand what I am looking at when I get to the example code snippet.